<a href="https://colab.research.google.com/github/KrisDA-1/Py.Skeletal-Variation/blob/main/Eskeletal_variation.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Using database we confirm that adults have 206 bones, using the following code

In [1]:
import pandas as pd
df=pd.read_csv('adult-human-skeleton.csv')
df #display it

name region subregion    side  fused_from
0                         frontal   head   cranium  center           2
1                   left parietal   head   cranium    left           1
2                  right parietal   head   cranium   right           1
3                   left temporal   head   cranium    left           1
4                  right temporal   head   cranium   right           1
..                            ...    ...       ...     ...         ...
201  right distal pedal phalanx 1   foot      toes   right           1
202  right distal pedal phalanx 2   foot      toes   right           1
203  right distal pedal phalanx 3   foot      toes   right           1
204  right distal pedal phalanx 4   foot      toes   right           1
205  right distal pedal phalanx 5   foot      toes   right           1

[206 rows x 5 columns]

**1st Hypothesis: 50% of bones are in hands and feet **

In [2]:
df['region'].value_counts()

region
hand     54
foot     52
torso    50
head     28
neck      8
leg       8
arm       6
Name: count, dtype: int64

In [3]:
(54+52)/206

0.5145631067961165

R//*Over 51% of bones are found in the hands and feet. Hypothesis is true*

**2nd Hyphotesis: Infants have more bones than adults**

In [4]:
df.sort_values(by='fused_from',ascending=False)

name region subregion    side  fused_from
40                        sternum  torso     chest  center           6
30                             c2   neck  vertebra  center           5
82                         sacrum  torso    pelvis  center           5
5                       occipital   head   cranium  center           4
83                         coccyx  torso    pelvis  center           4
..                            ...    ...       ...     ...         ...
201  right distal pedal phalanx 1   foot      toes   right           1
202  right distal pedal phalanx 2   foot      toes   right           1
203  right distal pedal phalanx 3   foot      toes   right           1
204  right distal pedal phalanx 4   foot      toes   right           1
205  right distal pedal phalanx 5   foot      toes   right           1

[206 rows x 5 columns]

This shows a wide variety of bone fusion across the body

In [5]:
##How many bones do infants have?
df['fused_from'].sum()

np.int64(305)

R//Human infants have 305 bones, many of which fuse together, resulting in 206 bones in the adult body.